embuilder.py

```diff
@@ -150,7 +152,7 @@ def build_port(port_name, lib_name, params):
 
 
 def static_library_name(name):
-  if shared.Settings.WASM_BACKEND:
+  if shared.Settings.WASM_BACKEND and shared.Settings.WASM_OBJECT_FILES:
```

Comment:
I think the extension should always be .a here, even in the LTO case.

Regardless of whether, the files inside are `.bc` or `.o` I think archives should always be called `.a`.  The exception is for fastcomp that doesn't use libraries at all but llvm-link'd bitcode.

Reply:
I don't feel strongly either way, but this PR is just matching the current behavior defined here: https://github.com/emscripten-core/emscripten/blob/incoming/tools/system_libs.py#L616